Make more options configurable

sebastianstigler opened the pull request:
In order to set the `width` of the slides (some beamer at work have only a low resolution), I made it and some other options configurabe like `theme`, `transition`and `start_slideshow_at`

damianavila commented on README.md:
```diff
@@ -126,6 +126,9 @@ With these options, your slides will get the `serif` theme and the
 
+There are also options for `controls`, `progress`, `history`, `width`,
+`minScale` and `slideNumber`
+
```
This is nice, can you add the height too?

damianavila commented on livereveal/main.js:
```diff
@@ -17,8 +17,14 @@ var config_section = new configmod.ConfigSection('livereveal',
+    controls: true,
+    progress: true,
+    history: true,
+    width: 1140,
```
Please add height too...

damianavila commented on livereveal/main.js:
```diff
@@ -17,8 +17,14 @@ var config_section = new configmod.ConfigSection('livereveal',
+    progress: true,
+    history: true,
+    width: 1140,
+    minScale: 1.0,
```
This sould be not modificable because of some codemirror issue, so I don't want to expose it... please remove it

adl replied:
Wouldn't it be enough to document that setting it to another value will break codemirror?  This way, those who do not use codemirror can still play with this setting.

As far as I am concerned, I have no usage for codemirror in presentation mode (I hide all code cells and do all the interactivity via widgets), but I've had many trouble to adapt my presentation to the width of the beamer (that I cannot know before hand) because of scaling issues.

damianavila replied:
OK, I did not think about people not showing code... the use case now makes sense... as far as it is well documented, at least with a note in the readme about the possibility to break codemirror, then I am ok to make it configurable...

sebastianstigler replied:
I did just that with the commit 233a92f.

damianavila commented:
Some minor comments... I will test it after the requested fixes... thanks!

damianavila commented:
Also, this will close https://github.com/damianavila/RISE/issues/97 :wink:

sebastianstigler commented:
Sorry,  I forget to post a comment to the second commit:

I have added height to the configurable options and added a warning for the use of the minScale option in the README.md and in the sourcecode.

damianavila commented:
>Sorry, I forget to post a comment to the second commit:

Yes I would love to have some kind of github ping when somebody commit to a PR... anyway, I will test on the weekend. If everything is OK (I presume that) I will merge it as is. Thanks!

damianavila commented on livereveal/main.js:
```diff
@@ -17,8 +17,15 @@ var config_section = new configmod.ConfigSection('livereveal',
+    progress: true,
+    history: true,
+    width: 1140,
+    height: 855, // 4:3 ratio
```
I would like to make this width 1024 and height 768, the deafult ones are really weird values.

damianavila commented:
Since we are modifying this values, I would like to use "more standard" standards... 1024 x 768 seems a good choice as default.

sebastianstigler commented:
Changed the default values for width and height as recomended

adl commented:
A few weeks ago, I tried changing 1140 to 1024 because I knew 1024 was the worst-case resolution for the beamer I would use, and ran into some scalling issue I did not understand.  And unfortunately I was too much in a hurry to investigate.  Maybe the main issue was that I started building the presentation with width=1140, and so it could not fit anymore with width=1024.  One thing I understood is that 1140 is special because it is actually the width of the notebook, so maybe it helps when moving back and forth between Reveal mode and notebook mode.  I ended-up keeping 1140 in my setup.

sebastianstigler commented:
How about I leave the `width=1140` and `height=855` in main.js and use the 1024x768 values as examples in the README.md?

damianavila commented:
I don't have too much problems with width 1140... in fact I selected that in the past because of the notebook width, as @adl suggested... let's then keep it as default with height 855... and I agree with @sebastianstigler about using 1024x768 in the README so people get the idea about the configurability of the parameter to a "standard" size...

@adl, about the scaling issue you saw... this is something weird beceause we mix things reveal calculate by itself and we also impose some of the notebook calculation of top of that... so probably the behaviour is buggy and we need more investigation in the future... in the meantime, I just want to provide people with config height and width options and the then browser zooming if something does not fit well...

sebastianstigler commented:
And here it is...

damianavila commented:
Thank you very much @sebastianstigler, merging now...
